middle_end/flambda2/to_cmm/to_cmm_env.ml

```diff
@@ -208,150 +282,313 @@ let get_exn_extra_args env k =
 
 (* Variable binding (for potential inlining). Also see [To_cmm_effects]. *)
 
-let is_inlinable_box effs ~extra =
-  (* [effs] is the effects and coeffects of some primitive operation, arising
-     either from the primitive itself or its arguments. If this is a boxing
-     operation (as indicated by [extra]) then we want to inline the box. However
-     this involves moving the arguments, so they must be pure (or at most have
-     generative effects, with no coeffects). *)
-  match (effs : Ece.t), (extra : extra_info option) with
-  | ((No_effects | Only_generative_effects _), No_coeffects), Some Boxed_number
+let next_order = ref (-1)
+
+let simple cmm_expr = Simple { cmm_expr }
+
+let splittable_primitive name args prim_effects make_expr =
+  Splittable { name; args; prim_effects; make_expr }
+
+let complex_no_split name cmm_expr effs =
+  splittable_primitive name [] effs (fun _ -> cmm_expr)
+
+let is_cmm_simple cmm =
```

Comment:
I think this should be an exhaustive match

Reply:
fixed